backend/census_historical_migration/workbooklib/excel_creation_utils.py

```diff
@@ -92,43 +87,32 @@ def map_simple_columns(wb, mappings, values):
         )
 
 
-def _census_date_to_datetime(cd):
-    lookup = {
-        "JAN": 1,
-        "FEB": 2,
-        "MAR": 3,
-        "APR": 4,
-        "MAY": 5,
-        "JUN": 6,
-        "JUL": 7,
-        "AUG": 8,
-        "SEP": 9,
-        "OCT": 10,
-        "NOV": 11,
-        "DEC": 12,
-    }
-    year = int(cd.split("-")[2])
-    month = lookup[cd.split("-")[1]]
-    day = int(cd.split("-")[0])
-    return date(year + 2000, month, day)
-
-
-# FIXME: Get the padding/shape right on the report_id
-def dbkey_to_test_report_id(Gen, dbkey):
-    g = Gen.select(Gen.audityear, Gen.fyenddate).where(Gen.dbkey == dbkey).get()
-    # month = g.fyenddate.split('-')[1]
-    # 2022JUN0001000003
-    # We start new audits at 1 million.
-    # So, we want 10 digits, and zero-pad for
-    # historic DBKEY report_ids
-    dt = _census_date_to_datetime(g.fyenddate)
-    return f"{g.audityear}-{dt.month:02}-TSTDAT-{dbkey.zfill(10)}"
+def add_hyphen_to_zip(zip):
+    strzip = str(zip)
+    if len(strzip) == 5:
+        return strzip
+    elif len(strzip) == 9:
+        return f"{strzip[0:5]}-{strzip[5:9]}"
+    else:
+        logger.info("ZIP IS MALFORMED IN WORKBOOKS E2E / SAC_CREATION")
```

Comment:
Possibly another place that should fail? Dunno.

Reply:
json-schema should catch this.

Reply:
For the future/future work: yes, but I think *all* Census zips lack the hyphen. So, we may end up wanting a xform that takes 9-digit zips and adds the hyphen, or lots of data won't migrate. In this case, it isn't that the data is "bad," it's just we might be expecting a different form. (Totally may be mis-remembering.)

I can't remember what we expect. I know this merged, but noting it here @sambodeme, @phildominguez-gsa .

Reply:
It looks like what you're describing, `... that takes 9-digit zips and adds the hyphen,` is what line 95 achieves. So either you meant to pull this into a separate xform for consistency, or you meant something else. I will ticket this and follow up with you.